### PR TITLE
Tidy of makefile for D4RII and movement of sources.

### PR DESCRIPTION
**Pull request by blckmn**
Created this as a proposal to make it easier to add more targets and clean up a few things.

I plan on adding target for XSR and X4R-SB - so that the smart port connection can be used with simple D8 telemetry (same as on the D4R) instead of the more complicated sport polling protocol.

This will allow sharing of the comm port on Flight Controllers with SBUS (using RX pin) and Telemetry (using TX pin).

### CODE REVIEW

**Comment by blckmn** [comment]
Note that some of the targets will fail, and any assistance on getting those other targets up and running with the src and obj directories for this PR would be great :)

**Comment by blckmn** [comment]
@fishpepper there is a missing file hal_dma.h (and c) for the rasp build.

**Comment by fishpepper** [comment]
Thanks! that looks very good! i will integrate it once you finished your work.

dma.*: i see you use a wildcard in your makefile. dma.c was never built in my makefile. dma.c/dma.h should be removed from src then. It is empty anyway ;)

**Comment by blckmn** [comment]
I'll do some updates on the bootloader too, and re-add that in another PR.

File: board/d4rii/target.mk [hunk 1]
```diff
@@ -1,152 +1,3 @@
...
-	if [ -a $(OPENOCD_PIDFILE) ]; then kill `cat $(OPENOCD_PIDFILE)`; fi;
-
-flash : $(RESULT).bin
-	if [ -a $(OPENOCD_PIDFILE) ]; then kill `cat $(OPENOCD_PIDFILE)`; fi;
-	st-flash write $(RESULT).bin 0x8000000
-
-.PHONY: board clean flash debug
+
+# continue with normal stm32f1 build
+include arch/stm32f1/stm32f1.mk
```
Comment (DieHertz) [inline board/d4rii/target.mk]: A newline at the end of file seems required

**Comment by blckmn** [comment]
@fishpepper this is pretty much ready - though will need to complete the boot loader changes too for those to work fully. Can you please test and give me any feed back?

**Comment by fishpepper** [comment]
Looking good so far. Thanks! I am not at home right now, i will give you feedback the next days.

**Comment by blckmn** [comment]
Sorry - didn't realise there was a merge conflict now :) fixed.

**Comment by fishpepper** [comment]
looks good. I created a branch for this:
https://github.com/fishpepper/OpenSky/tree/makefile_cleanup
I re-added the bootloader code.
I will now create a branch where i will add a stylechecker, this will take some time to get all the stylechecker complaints satisfied ;)

**Comment by fishpepper** [comment]
I finished the stylecheck stuff. See https://github.com/fishpepper/OpenSky/tree/stylecheck
This includes all your changes + bootloader + stylecheck cleanup (with some makefile tweaks)

**Comment by blckmn** [comment]
Cool. @fishpepper is there anything further you want me to do?

**Pull request merged**